python_backend/agents/base_agent.py

```
@@ -0,0 +1,326 @@
+        state["processing_metadata"]["end_time"] = time.time()
+        state["processing_metadata"]["total_duration"] = (
+            state["processing_metadata"]["end_time"] - 
+            state["processing_metadata"]["start_time"]
+        )
+        return state
+
+# Agent 팩토리
+class AgentFactory:
+    """Agent 인스턴스 생성 및 관리"""
+    
+    _agents: Dict[str, Type[BaseAgent]] = {}
```

Ruff (0.13.1)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)